autoscheduler/frontend/src/redux/reducers/schedules.ts

```diff
@@ -22,18 +24,77 @@ export interface ReplaceSchedulesAction {
     type: 'REPLACE_SCHEDULES';
     schedules: Meeting[][];
 }
-export type ScheduleAction = AddScheduleAction | RemoveScheduleAction | ReplaceSchedulesAction;
+export interface SaveScheduleAction {
+    type: 'SAVE_SCHEDULE';
+    index: number;
+}
+export interface UnsaveScheduleAction {
+    type: 'UNSAVE_SCHEDULE';
+    index: number;
+}
+export type ScheduleAction = AddScheduleAction | RemoveScheduleAction
+  | ReplaceSchedulesAction | SaveScheduleAction | UnsaveScheduleAction;
+
+interface Schedules {
+  schedules: Meeting[][];
+  savedSchedules: Meeting[][];
+}
+
+const initialSchedules: Schedules = {
+  schedules: [],
```

Comment:
Maybe rename this to `allSchedules` or something to be more specific about what it's doing

Reply:
Good idea, referencing `schedules.schedules` is a little confusing